--- ./external_modules/code/CAMB/forutils/Makefile_compiler	2024-06-13 20:38:40.778522420 -0400
+++ ./external_modules/code/CAMB2/forutils/Makefile_compiler	2024-06-11 18:28:16.856100094 -0400
@@ -6,12 +6,17 @@
 # F90DEBUGFLAGS: These flags are only used for creating Debug artefacts
 # F90RELEASEFLAGS: These flags are only used for creating Release artefacts
 
+#VM BEGINS
+#COCOA BEGINS
 # For standalone compiling set the compiler
-ifneq ($(COMPILER),gfortran)
-ifortErr = $(shell which ifort >/dev/null 2>&1; echo $$?)
-else
+#ifneq ($(COMPILER),gfortran)
+#ifortErr = $(shell which ifort >/dev/null 2>&1; echo $$?)
+#else
+#ifortErr = 1
+#endif
+#COCOA ENDS
+#VM ENDS
 ifortErr = 1
-endif
 
 ifeq "$(ifortErr)" "0"
 
@@ -51,14 +56,16 @@
 endif
 
 else
-
-major_version = $(shell gfortran -dumpversion 2>&1 | cut -d " " -f 3 | cut -d. -f 1)
-ifneq ($(shell test $(major_version) -gt 5; echo $$?),0)
-$(error gfortran version 6.3 or higher (or ifort 14+) is required)
-endif
-compiler_ver = $(shell gfortran -dumpversion 2>&1)
-
-F90C ?= gfortran
+#COCOA BEGINS
+#major_version = $(shell gfortran -dumpversion 2>&1 | cut -d " " -f 3 | cut -d. -f 1)
+#ifneq ($(shell test $(major_version) -gt 5; echo $$?),0)
+#$(error gfortran version 6.3 or higher (or ifort 14+) is required)
+#endif
+#compiler_ver = $(shell gfortran -dumpversion 2>&1)
+#F90C ?= gfortran
+F90C ?= $(FORTRAN_COMPILER)
+#COCOA ENDS
+#VM ENDS
 F90COMMONFLAGS ?= -cpp -ffree-line-length-none -fmax-errors=4 -MMD -fopenmp -fPIC
 F90DEBUGFLAGS ?= -g -O0 -fbacktrace
 F90RELEASEFLAGS ?= -O3 -ffast-math
