--- ./external_modules/code/CAMB/camb/_compilers.py	2024-06-13 20:38:40.049451000 -0400
+++ ./external_modules/code/CAMB2/camb/_compilers.py	2024-06-11 18:28:16.577356000 -0400
@@ -3,6 +3,7 @@
 import struct
 import platform
 import re
+from pkg_resources import parse_version
 
 is_windows = platform.system() == "Windows"
 
@@ -24,20 +25,21 @@
 def get_ifort_version():
     return call_command("ifort -v")
 
-
+#VM BEGINS
+#COCOA BEGINS
 def get_gfortran_version(command='gfortran'):
-    ver = call_command(command + " -dumpversion")
+    ver = call_command("$FORTRAN_COMPILER -dumpversion")
     if ver and '.' not in ver:
-        ver = call_command(command + " -dumpfullversion")
+        ver = call_command("$FORTRAN_COMPILER -dumpfullversion")
     return ver
-
+#COCOA ENDS
+#VM ENDS
 
 def check_ifort():
     return get_ifort_version() or False
 
 
 def check_gfortran(version=gfortran_min, msg=False, retry=False):
-    from packaging.version import parse as parse_version
     global compiler_environ
     gfortran_version = get_gfortran_version()
     version = str(version)
@@ -62,17 +64,13 @@
                         ver = x.split('-')[1]
                         if parse_version(best_version) <= parse_version(ver):
                             best_version = ver
-                            newpath = os.path.join(mingw, x)
-                            if os.path.exists(os.path.join(newpath, 'mingw64', 'bin')):
-                                newpath = os.path.join(newpath, 'mingw64', 'bin')
-                            elif os.path.exists(os.path.join(newpath, 'bin')):
-                                newpath = os.path.join(newpath, 'bin')
-                bin_path = os.path.join(mingw, 'bin')
-                if os.path.exists(bin_path):
-                    ver = get_gfortran_version('"' + os.path.join(bin_path, 'gfortran') + '"')
+                            newpath = os.path.join(mingw, x, 'mingw64', 'bin')
+                bin = os.path.join(mingw, 'bin')
+                if os.path.exists(bin):
+                    ver = get_gfortran_version('"' + os.path.join(bin, 'gfortran') + '"')
                     if ver and parse_version(best_version) <= parse_version(ver):
                         best_version = ver
-                        newpath = bin_path
+                        newpath = bin
         if newpath:
             if os.path.exists(os.path.join(newpath, 'gfortran.exe')):
                 if not compiler_environ["PATH"].startswith(newpath):
