--- makefile	2025-09-03 18:21:43
+++ makefile	2025-09-03 18:35:00
@@ -36,10 +36,15 @@
-ifeq ($(UNAME_S),Darwin)
-  LIBDYN = libcuba$(SUFFIX).dylib
-else
-  LIBDYN = libcuba$(SUFFIX).so
-endif
+ifeq ($(UNAME_S),Darwin)
+  LIBDYN  = libcuba$(SUFFIX).dylib
+  SOFLAGS = -dynamiclib -install_name @rpath/$(LIBDYN) -Wl,-undefined,dynamic_lookup
+else
+  LIBDYN  = libcuba$(SUFFIX).so
+  SOFLAGS = -shared -Wl,-soname,$(LIBDYN)
+endif
 
@@ -56,9 +61,7 @@
 $(LIBDYN): $(OBJS_U)
-	$(CC) -dynamiclib -o $@ $(OBJS_U) \
-	      -install_name @rpath/$(LIBDYN) \
-	      -Wl,-undefined,dynamic_lookup
+	$(CC) $(SOFLAGS) -o $@ $(OBJS_U) $(LIBS)
