--- InstallableLikelihood.py	2026-05-04 21:24:37.094013236 +0000
+++ InstallableLikelihood2.py	2026-05-04 21:24:37.101867689 +0000
@@ -40,19 +40,19 @@
             old = False
-            try:
-                installed = self.is_installed(path=packages_path)
-            except Exception as excpt:  # catches VersionCheckError and unexpected ones
-                installed = False
-                old = isinstance(excpt, VersionCheckError)
-                logger.error(f"{type(excpt).__name__}: {excpt}")
-            if not installed:
-                not_or_old = (
-                    "is not up to date" if old else "has not been correctly installed"
-                )
-                raise ComponentNotInstalledError(
-                    logger,
-                    (
-                        f"The data for this likelihood {not_or_old}. To install it, "
-                        f"run `cobaya-install {name}{' --upgrade' if old else ''}`"
-                    ),
-                )
+            #try:
+            #    installed = self.is_installed(path=packages_path)
+            #except Exception as excpt:  # catches VersionCheckError and unexpected ones
+            #    installed = False
+            #    old = isinstance(excpt, VersionCheckError)
+            #    logger.error(f"{type(excpt).__name__}: {excpt}")
+            #if not installed:
+            #    not_or_old = (
+            #        "is not up to date" if old else "has not been correctly installed"
+            #    )
+            #    raise ComponentNotInstalledError(
+            #        logger,
+            #        (
+            #            f"The data for this likelihood {not_or_old}. To install it, "
+            #            f"run `cobaya-install {name}{' --upgrade' if old else ''}`"
+            #        ),
+            #    )
         super().__init__(*args, **kwargs)
