--- mflike.py	2025-03-30 15:58:46.562749777 -0400
+++ mflike2.py	2025-03-30 16:02:27.645027950 -0400
@@ -45,10 +45,16 @@
 from numbers import Real
 import sacc
 from cobaya.conventions import data_path, packages_path_input
-from cobaya.likelihoods.base_classes import InstallableLikelihood
+#VM BEGINS
+#from cobaya.likelihoods.base_classes import InstallableLikelihood
+from cobaya.likelihood import Likelihood
+#VM ENDS
 from cobaya.log import LoggedError
 
-class _MFLike(InstallableLikelihood):
+#VM BEGINS
+#class _MFLike(InstallableLikelihood):
+class _MFLike(Likelihood):
+#VM ENDS
     _url = "https://portal.nersc.gov/cfs/sobs/users/MFLike_data"
     _release = "v0.8"
     install_options = {
