--- mflike.py	2024-07-11 20:47:37.909715210 -0400
+++ mflike2.py	2024-07-11 20:50:22.611406000 -0400
@@ -24,14 +24,20 @@
 
 import numpy as np
 from cobaya.conventions import data_path, packages_path_input
-from cobaya.likelihoods.base_classes import InstallableLikelihood
+#VM BEGINS
+#from cobaya.likelihoods.base_classes import InstallableLikelihood
+from cobaya.likelihood import Likelihood
+#VM ENDS
 from cobaya.log import LoggedError
 from cobaya.tools import are_different_params_lists
 
 from .theoryforge import TheoryForge
 
 
-class MFLike(InstallableLikelihood):
+#VM BEGINS
+#class MFLike(InstallableLikelihood):
+class MFLike(Likelihood):
+#VM ENDS
     _url = "https://portal.nersc.gov/cfs/sobs/users/MFLike_data"
     _release = "v0.8"
     install_options = {"download_url": f"{_url}/{_release}.tar.gz"}
