--- model.py    2026-05-04 22:31:13.351043922 +0000
+++ model2.py   2026-05-04 22:31:20.752892972 +0000
@@ -1211,10 +1211,17 @@
             if len(agnostic_likes) > 1 and not all(assign.values()):
-                raise LoggedError(
-                    self.log,
-                    "More than one parameter-agnostic likelihood/theory "
-                    "with respect to %s: %r. Cannot decide "
-                    "parameter assignments.",
-                    option,
-                    agnostic_likes,
-                )
+                #VM BEGINS (ALLOW MULTIPLE EMULATORS AS THEORY)
+                #raise LoggedError(
+                #    self.log,
+                #    "More than one parameter-agnostic likelihood/theory "
+                #    "with respect to %s: %r. Cannot decide "
+                #    "parameter assignments.",
+                #    option,
+                #    agnostic_likes,
+                #)
+                for i in range(len(agnostic_likes)):
+                    component = agnostic_likes[i]
+                    for p, assigned in assign.items():
+                        #if not assigned:
+                        assign[p] += [component]
+                #VM ENDS
             elif agnostic_likes:  # if there is only one