--- act_dr6_cmbonly.py	2025-03-30 11:30:52.999816797 -0400
+++ act_dr6_cmbonly2.py	2025-03-30 11:39:52.591766644 -0400
@@ -12,7 +12,7 @@
     url: str = "https://lambda.gsfc.nasa.gov/data/act/pspipe/sacc_files/"
     file_base_name: str = "dr6_data_cmbonly"
     input_file: str = f"{file_base_name}.fits"
-    data_folder: str = "ACTDR6CMBonly"
+    data_folder: str = "act_dr6_cmbonly"
     version: str = "v1.0"
 
     install_options: dict = {
@@ -43,16 +43,22 @@
     }
 
     def initialize(self):
-        if self.packages_path is None:
-            self.data_folder = os.path.join(
-                os.path.dirname(os.path.realpath(__file__)),
-                "data")
-        else:
-            from cobaya.conventions import data_path
-            self.data_folder = os.path.join(
-                self.packages_path,
-                data_path,
-                self.data_folder)
+        #if self.packages_path is None:
+        #    self.data_folder = os.path.join(
+        #        os.path.dirname(os.path.realpath(__file__)),
+        #        "data")
+        #else:
+        #    from cobaya.conventions import data_path
+        #    self.data_folder = os.path.join(
+        #        self.packages_path,
+        #        data_path,
+        #        self.data_folder)
+        
+        self.data_folder = os.path.join(
+                             os.environ['ROOTDIR'],
+                             "/external_modules/data/",
+                             self.data_folder
+                           )
 
         import sacc
         input_filename = os.path.join(self.data_folder, self.version,
